hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/impl/ITestS3APutIfMatch.java

```diff
@@ -0,0 +1,564 @@
+        // conditional_write, conditional_write_statistics should remain 0
+        try (FSDataOutputStream outputStream = fs.create(testFile, true)) {
+            outputStream.write(SMALL_FILE_BYTES);
+            updateStatistics(outputStream);
+        }
+        conditionalCreate = statistics.lookupCounterValue(Statistic.CONDITIONAL_CREATE.getSymbol());
+        conditionalCreateFailed = statistics.lookupCounterValue(Statistic.CONDITIONAL_CREATE_FAILED.getSymbol());
+        assertEquals("Write with overwrite=true: conditional_create should be 0", 0L, conditionalCreate);
+        assertEquals("Write with overwrite=true: conditional_create_failed should be 0", 0L, conditionalCreateFailed);
+
+        // write in path where file already exists with overwrite = false
+        // conditional_write, conditional_write_statistics should remain 0
+        try (FSDataOutputStream outputStream = fs.create(testFile, false)) {
+            outputStream.write(SMALL_FILE_BYTES);
+            updateStatistics(outputStream);
+        } catch (FileAlreadyExistsException e) {}
+        conditionalCreate = statistics.lookupCounterValue(Statistic.CONDITIONAL_CREATE.getSymbol());
+        conditionalCreateFailed = statistics.lookupCounterValue(Statistic.CONDITIONAL_CREATE_FAILED.getSymbol());
+        assertEquals("Write in path where file already exists with overwrite=false: conditional_create should be 0", 0L, conditionalCreate);
+        assertEquals("Write in path where file already exists with overwrite=false: conditional_create_failed should be 0", 0L, conditionalCreateFailed);
+
+        // delete the file
+        fs.delete(testFile, false);
+
+        // write in path where file doesn't exist with overwrite = false
+        // conditional_write, conditional_write_statistics should remain 0
+        try (FSDataOutputStream outputStream = fs.create(testFile, false)) {
+            outputStream.write(SMALL_FILE_BYTES);
+            updateStatistics(outputStream);
+        }
+        conditionalCreate = statistics.lookupCounterValue(Statistic.CONDITIONAL_CREATE.getSymbol());
+        conditionalCreateFailed = statistics.lookupCounterValue(Statistic.CONDITIONAL_CREATE_FAILED.getSymbol());
+        assertEquals("Write in path where file does not exist with overwrite=false: conditional_create should be 0", 0L, conditionalCreate);
+        assertEquals("Write in path where file does not exist with overwrite=false: conditional_create_failed should be 0", 0L, conditionalCreateFailed);
+    }
+
+    @Test
+    public void testConditionalWriteStatisticsWithIfNoneMatch() throws Throwable {
+        FileSystem fs = getFileSystem();
+        Path testFile = methodPath();
+
+        FSDataOutputStream stream = getStreamWithFlags(fs, testFile, true, null, false);
+        updateStatistics(stream);
+        stream.write(SMALL_FILE_BYTES);
+        stream.close();
+
+        long conditionalCreate = statistics.lookupCounterValue(Statistic.CONDITIONAL_CREATE.getSymbol());
```

Comment:
There's a class IOStatisticAssertions to help assert here...look at its usagesj. It's assertions result in AssertJ assertions, so you can go

```java
verifyStatisticCounterValue(statistics, CONDITIONAL_CREATE.getSymbol, 1)
```

same elsewhere